File: Practise/1200/C_Book_Reading.py
# https://codeforces.com/problemset/problem/1213/C

def main():
    # read test cases
    t = int(input())
    for _ in range(t):
        # read n and m (n : total pages, m : divisible by)
        n, m = map(int, input().split())
        # everytime reach a page divisible by m, jot down the last digit of the page number
        # and add it to the sum
        # the sum is the answer
        # get the cycle length
        cycle = 0
        for i in range(1, m + 1):
            if (i * i) % m == 0:
                cycle = i
                break
        # get the number of cycles
        cycles = n // m // cycle
        # get the sum of the last digits of the pages in the last cycle
        last_cycle = n // m % cycle
        sum_last_digits = 0
        # for cycle less than 1, calculate the sum of the last digits
        if cycle <= 1:
            for i in range(1, n+1):
                if i % m == 0:
                    sum_last_digits += i % 10
        # for cycle greater than 1, calculate the sum of the last digits in one cycle
        # and multiply it by the number of cycles (except the last cycle)
        else:
            for i in range(1, cycle+1):
                if i % m == 0:
                    sum_last_digits += i % 10
            sum_last_digits *= cycles
            # add the sum of the last digits in the last cycle
            for i in range(1, last_cycle+1):
                if i % m == 0:
                    sum_last_digits += i % 10
                    
        ans = cycles * sum_last_digits
        # print the answer
        print(ans)
        
        
if __name__ == '__main__':
    main()
    

File: Practise/1200/C_Book_Reading.cpp
#include <bits/stdc++.h>
#define forn(i, n) for (int i = 0; i < int(n); i++)
using namespace std;
int main() {
    // read number of test cases
    int q;
    cin >> q;

    // iterate over test cases
    forn(i, q) {
        // read n and m
        long long n, m;
        cin >> n >> m;
        // calculate number of digits in n
        n = n / m;
        // store sum of digits in each position
        vector<int> digits(10);
        forn(i, 10)
            digits[i] = ((i + 1) * m) % 10;
        // calculate sum of digits in positions 1 to n % 10
        long long sum = 0;
        forn(i, n % 10)
            sum += digits[i];
        // calculate sum of digits in positions 1 to 10
        // multiply with the number of times 1 to 10
        // appear in positions 1 to n
        cout << sum + n / 10 * accumulate(digits.begin(), digits.end(), 0LL) << endl;
    }
}